ui/src/vayla-design-lib/datepicker/datepicker.tsx

```diff
@@ -6,13 +6,78 @@ import { fi } from 'date-fns/locale';
 import { IconColor, Icons, IconSize } from 'vayla-design-lib/icon/Icon';
 import { createClassName } from 'vayla-design-lib/utils';
 import { TextField, TextInputIconPosition } from 'vayla-design-lib/text-field/text-field';
+import { CloseableModal } from 'vayla-design-lib/closeable-modal/closeable-modal';
+import { formatDateShort } from 'utils/date-utils';
+import { parse, isValid } from 'date-fns';
 import { useCloneRef } from 'utils/react-utils';
 
 type DatePickerProps = {
     value: Date | undefined;
-    onChange?: (date: Date | undefined) => void;
+    onChange: (date: Date | undefined) => void;
+    wide?: boolean;
 } & Omit<React.InputHTMLAttributes<HTMLInputElement>, 'value' | 'onChange'>;
 
+type DatePickerInputProps = {
+    openDatePicker: () => void;
+    date: Date | undefined;
+    setDate: (date: Date | undefined) => void;
+    wide: boolean | undefined;
+} & Omit<React.InputHTMLAttributes<HTMLInputElement>, 'value' | 'onChange'>;
+
+const DatePickerInput = React.forwardRef<HTMLInputElement, DatePickerInputProps>(
+    ({ openDatePicker, date, setDate, wide, ...props }, ref) => {
+        const [value, setValue] = React.useState<string>('');
+        const localRef = useCloneRef<HTMLInputElement>(ref);
+        React.useEffect(() => {
+            if (document.activeElement !== localRef.current) {
+                setValue(date ? formatDateShort(date) : '');
+            }
+        }, [date]);
```

Comment:
Tämä vaatinee vähän selitystä. Yleisesti ottaen halutaan, että tekstikentän arvo seuraa valittua päivämäärää ja toisin päin. Tuo sidonta päivämäärästä tekstiarvoon tehdään tällä. Tämä kuitenkin aiheuttaa normaalisti uudelleenpiirron aina kun päivämäärä vaihtuu alta, joka aiheuttaa käyttäjän kannalta todella ikävää pomppimista -> estetään tällä uudelleenpiirto mikäli focus on tässä kentässä.